automation_tools/repository.py

```diff
@@ -0,0 +1,126 @@
+"""Taks for managing repositories"""
+from __future__ import print_function
+
+import sys
+
+from automation_tools.utils import distro_info
+from fabric.api import put, run
+
+if sys.version_info[0] == 2:
+    from StringIO import StringIO  # (import-error) pylint:disable=F0401
+else:
+    from io import StringIO
+
+
+def disable_repos(*args, **kwargs):
+    """Disable repos passed as ``args``. Custom repositories can be disabled
+    through the ``kwargs``, see the example below for more information.
+
+    For example::
+
+        disable_repos('repo1', 'repo2', repo3=False, repo4=True)
+
+    The above call will disable ``repo1`` and ``repo2`` using
+    ``subscription-manager repos --disable``. The ``repo3`` and ``repo4``
+    arguments will be considered custom repositories. For ``repo3`` this task
+    will disable it but will not delete de repository file, but for ``repo4``
+    it will delete de repository file.
+
+    """
+    run(
+        'subscription-manager repos --disable {0}'
+        .format(' '.join(['"{0}"'.format(repo) for repo in args]))
+    )
+
+    for name, delete in kwargs.items():
+        repo_file = '/etc/yum.repos.d/{0}.repo'.format(name)
+        if delete:
+            run('rm {0}'.format(repo_file))
+        else:
+            run('sed -i -e "s/^enabled=.*/enabled=0/" {0}'.format(repo_file))
+
+
+def enable_repos(*args, **kwargs):
+    """Enable repos passed as ``args`` and create one repofile with each
+    ``kwargs``.
+
+    For example::
+
+        manage_repos('repo1', custom_repo='http://repourl.domain.com/path')
+
+    The above call will result in enabling ``repo1`` using
+    ``subscription-manager repos --enable``. Also will create a
+    custom_repo.repo file with the following contents::
+
+        [custom_repo]
+        name=custom_repo
+        baseurl=http://repourl.domain.com/path
+        enabled=1
+        gpgcheck=0
+
+    """
+    run('subscription-manager repos --enable {0}'
+        .format(' '.join(['"{0}"'.format(repo) for repo in args])))
+
+    for name, url in kwargs.items():
+        repo_file = StringIO()
+        repo_file.write(
+            '[{name}]\n'
+            'name={name}\n'
+            'baseurl={url}\n'
+            'enabled=1\n'
+            'gpgcheck=0\n'
+            .format(name=name, url=url)
+        )
+        put(local_path=repo_file,
+            remote_path='/etc/yum.repos.d/{0}.repo'.format(name))
+        repo_file.close()
+
+
+def enable_satellite_repos(cdn=False, disable_enabled=True):
+    """Enable repositories required to install Satellite 6
+
+    If ``disable_enabled`` is ``True``, then this task will first disable any
+    already enabled repository, including beaker ones.
+
+    """
+    if isinstance(cdn, str):
+        cdn = (cdn.lower() == 'true')
+    if isinstance(disable_enabled, str):
+        disable_enabled = (disable_enabled.lower() == 'true')
+
+    if disable_enabled:
```

Comment:
Using an explicit test like `if disable_enabled is True` may prevent some subtle bugs.